pkg/chunk/cache/fifo_cache.go

```diff
@@ -149,125 +170,106 @@ func (c *FifoCache) Fetch(ctx context.Context, keys []string) (found []string, b
 		}
 
 		found = append(found, key)
-		bufs = append(bufs, val.([]byte))
+		bufs = append(bufs, val)
 	}
-
 	return
 }
 
 // Store implements Cache.
 func (c *FifoCache) Store(ctx context.Context, keys []string, bufs [][]byte) {
-	values := make([]interface{}, 0, len(bufs))
-	for _, buf := range bufs {
-		values = append(values, buf)
-	}
-	c.Put(ctx, keys, values)
+	c.Put(ctx, keys, bufs)
```

Comment:
Having both `Store` and `Put` methods seems redundant. We need `Store`, since it's part of `cache.Cache` interface, but maybe we can remove `Put`?

Reply:
agree